### DISCUSSION TITLE
Trusted API Check

### DISCUSSION THREAD

**Discussion by Spazzinq** [GriefPrevention/GriefPrevention#1542] [Q&A]
I have dug through the GriefPrevention API and have not been able to find a de facto method to check if a player is /trusted in a claim. Could someone nudge me in the correct direction?

**Comment by Jikoo** [comment]
You're probably after `Claim#checkPermission(Player, ClaimPermission, Event)`. Note that this is pretty new API and isn't in official releases yet, so if you want compatibility for users who aren't running dev builds you probably want to use a solution more like [this](https://github.com/TechFortress/GriefPrevention/issues/1438#issuecomment-872363793).

If the player you're trying to check is not online or you explicitly only want to check the trust list and nothing else, you can use `Claim#hasExplicitPermission(UUID, ClaimPermission)` for explicit trust, though since offline players do not support permissions, the permission-based trusts will not be available.
